sink/test-logic

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2017 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+# Fixtures
+CONTAINS = [
+    # Basic comparisons
+    { "ret": True,  "expect": True, "result": True },
+    { "ret": False, "expect": 1, "result": 2 },
+    { "ret": False, "expect": "x", "result": "2" },
+    { "ret": True,  "expect": "x", "result": "x" },
+    { "ret": False, "expect": "x", "result": None },
+    { "ret": False, "expect": "x", "result": { } },
+    { "ret": False, "expect": "x", "result": [ ] },
+    { "ret": False, "expect": { }, "result": [ ] },
+    { "ret": False, "expect": { }, "result": "x" },
+    { "ret": False, "expect": [ ], "result": "x" },
+
+    # Lists
+    { "ret": True,  "expect": [ "a" ], "result": [ "a", "b", "x", "z" ] },
+    { "ret": True,  "expect": [ "a", "b" ], "result": [ "a", "b", "x", "z" ] },
+    { "ret": False, "expect": [ "y" ], "result": [ "a", "b", "x", "z" ] },
+    { "ret": False, "expect": [ "z" ], "result": [ "a", "b", "x", "z" ] },
```

Comment:
This reflects what I mentioned above -- `z` is clearly contained in the list, so "contains" is a poor name.